src/Engines/MeilisearchEngine.php

```diff
@@ -172,11 +172,11 @@ public function mapIds($results)
      */
     public function map(Builder $builder, $results, $model)
     {
-        if (is_null($results) || 0 === count($results['hits'])) {
+        if (is_null($results) || 0 === $results->getHitsCount()) {
```

Comment:
I would suggest to check if the `$results` parameter is a `SearchResult` object before and then casting the object to an array.

e.g.

```php
if ($results instanceof SearchResult) {
    $results = $results->toArray();
}
```

Reply:
Correct me if I'm wrong, but if in the custom search we use `rawSearch()` instead of `search()` we can keep `count($results['hits'])` instead of `->getHitsCount()` ?

Reply:
@shokme correct. but I think as you mentioned in https://github.com/meilisearch/meilisearch-laravel-scout/issues/87#issuecomment-770201231 that the PHP SDK (I didn't check on this) changed, it may be that other integrations are broken or will break if updated to the newer versions.

With my change we could also fix older integrations that copied the example from the docs with `->search()` in their custom query.

Reply:
@shokme the only change needed to work with the new behaviour would be my suggestion from https://github.com/meilisearch/meilisearch-laravel-scout/pull/86#discussion_r567229784

Reply:
I don't remember everything about the PHP SDK changes, but it safe as long as it allow us to work with arrays and as I remember it will.

if we do this change, we should do it for every case (I didn't check if there is more than just that one) that can throw the object as array error.
We should find a away to have the `->toArray()` conversion earlier than in each methods ?

Anyway, it is hard to me to find time at the moment. So I can't work on making everything perfect right now.
If you have free time and want to do it feel free to open **another** PR that will apply your change on every case that could throw an object as array error.

Thank you for your time !